aea/package_manager/base.py

```diff
@@ -263,6 +290,16 @@ def add_package(self, package_id: PackageId) -> "BasePackageManager":
             str(package_id.package_type), package_id.public_id, dest=str(download_path)
         )
 
+        if with_dependencies:
+            for dependency_package_id in self.get_package_dependencies(
+                package_id=package_id
+            ):
+                if self.get_package_config_file(dependency_package_id).exists():
+                    config = self._get_package_configuration(package_id=package_id)
+                    if config.package_id != dependency_package_id:
+                        self.update_package(dependency_package_id)
```

Comment:
I think we should make this check more strict regards to the package hash. We don't manage packages in the `packages/` directory using the package version, we use package hashes for them. So I would suggest we compare the package hashes rather to check if we want to update the package or add the package. 

Also it would be ideal if we raise an error if we're updating the package because the user should know that there was a change in the third party package